docs/querying/sql.md

```diff
@@ -629,7 +629,9 @@ Druid SQL uses four different native query types.
 period)`, have no other grouping expressions, no HAVING or LIMIT clauses, no nesting, and either no ORDER BY, or an
 ORDER BY that orders by same expression as present in GROUP BY. It also uses Timeseries for "grand total" queries that
 have aggregation functions but no GROUP BY. This query type takes advantage of the fact that Druid segments are sorted
-by time.
+by time. The SQL planner can inject `timestampResultField` in the query context when there is a grouping key. This
```

Comment:
Btw, the rest of this patch LGTM, other than the doc stuff mentioned here.

Reply:
OK, I agree. It could be better to make it as a user-facing configuration before documenting it. I think it can be more structured such as an explicit field in timeseries query spec to access timestamp instead of query context field if it should be a user-facing configuration.